python/lib/client/dmod/client/request_clients.py

```diff
@@ -505,6 +625,43 @@ async def download_from_dataset(self, dataset_name: str, item_name: str, dest: P
                     if not has_data:
                         return message_object
 
+    async def get_serialized_datasets(self, dataset_name: Optional[str] = None) -> Dict[str, dict]:
+        """
+        Get dataset objects in serialized form, either for all datasets or for the one with the provided name.
+
+        Parameters
+        ----------
+        dataset_name : Optional[str]
+            The name of a specific dataset to get serialized details of, if only one should be obtained.
+
+        Returns
+        -------
+        Dict[str, dict]
+            A dictionary, keyed by dataset name, of serialized dataset objects.
+        """
+        # TODO: may need to generalize this and add to super class
+        if dataset_name is None:
+            datasets = await self.list_datasets()
+        else:
+            # TODO: improve how this is use so that it can be safely, efficiently put everywhere it **may** be needed
+            await self._async_acquire_session_info()
+            datasets = [dataset_name]
+        serialized = dict()
+        action = ManagementAction.QUERY
+        query = DatasetQuery(query_type=QueryType.GET_SERIALIZED_FORM)
+        try:
+            for d in datasets:
+                request = MaaSDatasetManagementMessage(action=action, query=query, dataset_name=d,
+                                                       session_secret=self.session_secret)
+                self.last_response: DatasetManagementResponse = await self.async_make_request(request)
+                if self.last_response.success:
+                    serialized[d] = self.last_response.data[DatasetManagementResponse._DATA_KEY_QUERY_RESULTS]
+                # TODO: what to do if any are not successful
```

Comment:
I think we can avoid this more or less, or at least have less of an opinion about it if we move this method to an async generator rather than an accumulator. We could just yield results as we get them.